tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/GrpcDotNet/GrpcAspNetCoreServer/GrpcDotNetServerCommon.cs

```diff
@@ -0,0 +1,84 @@
+﻿// <copyright file="GrpcDotNetServerCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+#if !NET461
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Headers;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.Tagging;
+using Microsoft.AspNetCore.Http;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Grpc.GrpcDotNet.GrpcAspNetCoreServer
+{
+    internal static class GrpcDotNetServerCommon
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(GrpcDotNetServerCommon));
+
+        public static Scope? CreateServerSpan<T>(Tracer tracer, T instance, HttpRequest requestMessage)
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId.Grpc))
+            {
+                return null;
+            }
+
+            Scope? scope = null;
+            var method = instance.DuckCast<ServerCallHandlerBaseStruct>().MethodInvoker.Method;
+            try
+            {
+                var tags = new GrpcServerTags();
+                GrpcCommon.AddGrpcTags(tags, tracer, method.GrpcType, name: method.Name, path: method.FullName, serviceName: method.ServiceName);
+
+                // If we have a local span (e.g. from aspnetcore) then use that as the parent
+                // Otherwise, use the distributed context as the parent
+                var spanContext = tracer.ActiveScope?.Span.Context;
+                if (spanContext is null)
+                {
+                    spanContext = ExtractPropagatedContext(requestMessage);
+                }
+
+                var serviceName = tracer.DefaultServiceName ?? "grpc-server";
+
+                scope = tracer.StartActiveInternal(GrpcCommon.OperationName, parent: spanContext, tags: tags, serviceName: serviceName);
+
+                var span = scope.Span;
+                span.Type = SpanTypes.Grpc;
+                span.ResourceName = method.FullName;
+                tracer.TracerManager.Telemetry.IntegrationGeneratedSpan(IntegrationId.Grpc);
```

Comment:
Are we missing the App Analytics settings here? They are deprecated but should continue working for now.

Reply:
This is set in `GrpcCommon.AddGrpcTags()` (called at the start of this method): https://github.com/DataDog/dd-trace-dotnet/blob/andrew/grpc-dotnet/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/GrpcCommon.cs#L46

Reply:
Thanks!